libs/balances-and-allowances/src/hooks/useCurrencyAmountBalance.ts

```diff
@@ -0,0 +1,22 @@
+import { useMemo } from 'react'
+
+import { TokenWithLogo } from '@cowprotocol/common-const'
+import { CurrencyAmount } from '@uniswap/sdk-core'
+
+import { useTokensBalances } from './useTokensBalances'
+
+export function useCurrencyAmountBalance(
+  token: TokenWithLogo | undefined | null
```

Comment:
We actually have a generic type `Nullish` but it is in cowswap app directory.
Probably we should extract it to an independent library.
I would prefer keep both, because there are cases where we have `undefined` and `null `